test/test_pheromone.py

```diff
@@ -29,4 +28,36 @@ def test_get_pheromone_level():
     pheromone.leave_pheromone(pos, pheromone_status)
     levels = pheromone.get_pheromone_level(pos)
 
-    assert levels == {'coming from colony': 0, 'coming from food': pheromone_status}
\ No newline at end of file
+    assert levels == {'coming from colony': 0, 'coming from food': pheromone_status}
+
+def test_reduce_pheromones():
+    grid_shape = (10, 10)
+    pheromone = Pheromone(grid_shape)
+    
+    pos = (5, 5)
+
+    # Ant coming from food
+    pheromone_status = 1
+    pheromone.leave_pheromone(pos, pheromone_status)
+
+    # Ant coming from colony
+    pheromone_status = - 1
+    pheromone.leave_pheromone(pos, pheromone_status)
+
+    # First reduction
+    pheromone.reduce_pheromones(reducing_factor = 0.5)
+    levels = pheromone.get_pheromone_level(pos)
+    assert levels == {'coming from colony': - 0.5, 'coming from food': 0.5}
+
+    # Second reduction
+    pheromone.reduce_pheromones(reducing_factor = 0.5)
+    levels = pheromone.get_pheromone_level(pos)
+    assert levels == {'coming from colony': - 0.25, 'coming from food': 0.25}
+
+    # Further reductions
+    for _ in range(5):
+        pheromone.reduce_pheromones(reducing_factor = 0.5)
```

Comment:
instead of for loop: 
```suggestion
        pheromone.reduce_pheromones(reducing_factor = 0.5**5)
```

Reply:
Thanks for this improvement, I didn't think of this solution at the time, but I knew there was something better than the for-loop!